### PR TITLE
Refactoring 

### PR DESCRIPTION
**Pull request by alimertkoc**

### CODE REVIEW

File: Part1/code/main.py [hunk 1]
```diff
@@ -1,47 +1,42 @@
...
 
-dataset = STL10('./data', split='train', download=True)
-# Create the VisualAcuityDataset with a specific month_age
-month_ages = list(range(0, 13))  # Create a list of ages from 0 to 12
-va_datasets = []
-lc_datasets=[]
-for month_age in month_ages:
-    va_datasets.append(VisualAcuityDataset(dataset, month_age=month_age))
-    lc_datasets.append(LimitedColorPerceptionDataset(dataset, month_age=month_age))
+# STL10 dataset by Stanford University (https://cs.stanford.edu/~acoates/stl10/)
```
Comment (alimertkoc) [inline Part1/code/main.py:8]: the origin is maybe true, but the place we get the dataset is torchvision.datasets, not cs.stanford.edu

File: Part1/code/VisualAcuityDataset.py [hunk 2]
```diff
@@ -1,73 +1,58 @@
...
+
 
 class VisualAcuityDataset(Dataset):
-    def __init__(self, dataset, month_age, max_age=5.0, max_sigma=5.0):
+    def __init__(self, dataset, month_age, max_age=5, max_sigma=5):
         """
-        Initializes the dataset with flexible visual acuity.
-
         Args:
-            dataset (Dataset): The original dataset to wrap.
```
Comment (alimertkoc) [inline Part1/code/VisualAcuityDataset.py:15]: since dataset is an argument to this function, I think it still need to be stay.

File: Part1/code/LimitedColorPerception.py [hunk 3]
```diff
@@ -1,58 +1,66 @@
...
 
+
 class LimitedColorPerceptionDataset(Dataset):
-    def __init__(self, dataset, month_age, max_age=12.0):
+    def __init__(self, dataset, month_age, max_age=6) -> None:
         """
-        Initializes the dataset with limited color perception based on infant color development.
-
         Args:
-            dataset (Dataset): The original dataset to wrap.
```
Comment (alimertkoc) [inline Part1/code/LimitedColorPerception.py:14]: this dataset arg, too.

File: Part1/code/LimitedColorPerception.py [hunk 4]
```diff
@@ -1,58 +1,66 @@
...
-            4: {'red': 0.95, 'green': 0.85, 'blue': 0.7, 'contrast': 0.8},
-            5: {'red': 0.98, 'green': 0.9, 'blue': 0.75, 'contrast': 0.85},
-            6: {'red': 1.0, 'green': 0.95, 'blue': 0.8, 'contrast': 0.9},
-            7: {'red': 1.0, 'green': 0.97, 'blue': 0.85, 'contrast': 0.95},
-            8: {'red': 1.0, 'green': 1.0, 'blue': 0.9, 'contrast': 1.0},
-            9: {'red': 1.0, 'green': 1.0, 'blue': 1.0, 'contrast': 1.0},
-            10: {'red': 1.0, 'green': 1.0, 'blue': 1.0, 'contrast': 1.0},
-            11: {'red': 1.0, 'green': 1.0, 'blue': 1.0, 'contrast': 1.0},
-            12: {'red': 1.0, 'green': 1.0, 'blue': 1.0, 'contrast': 1.0},
+        # Inital color properties (red, green, blue, and contrast)
```
Comment (alimertkoc) [inline Part1/code/LimitedColorPerception.py]: if the results are similar, I think this is a more cool way to implement. if not so, we can revert the changes since the cost is high.

File: Part1/code/main.py (hunk 1 above)
Comment (sahinozan) [reply Part1/code/main.py:8]: Torch is the environment who provides the dataset which we are not interested in. However, the dataset's owner and provider is still Stanford University.

File: Part1/code/VisualAcuityDataset.py (hunk 2 above)
Comment (sahinozan) [reply Part1/code/VisualAcuityDataset.py:15]: Re-added the dataset argument to the docstring.

File: Part1/code/LimitedColorPerception.py (hunk 3 above)
Comment (sahinozan) [reply Part1/code/LimitedColorPerception.py:14]: Re-added the dataset argument to the docstring.

File: Part1/code/LimitedColorPerception.py (hunk 4 above)
Comment (sahinozan) [reply Part1/code/LimitedColorPerception.py]: The cost is irrelevant, since manually creating a fixed-size dictionary is practically the same with doing it via for loop. Especially for a dictionary of a size between 6 and 12. I suggest keeping the new snippet, since the final version of this part should be provided by Rasul (considering that he is the one who provided images to the report with his own channel and contrast configuration). After merging this to the main branch, this part will be overridden by Rasul's branch.
